qadence/backends/pulser/convert_ops.py

```diff
@@ -39,13 +40,14 @@ def forward(
         state: torch.Tensor,
         values: dict[str, torch.Tensor] = dict(),
         qubit_support: tuple | None = None,
+        noise: Noise | None = None,
         endianness: Endianness = Endianness.BIG,
     ) -> torch.Tensor:
-        if self.block.is_parametric:
-            block_mat = block_to_tensor(
-                self.block, values, qubit_support=qubit_support, endianness=endianness
-            ).squeeze(0)
-        else:
-            block_mat = self.block_mat
-
-        return torch.sum(torch.matmul(state, block_mat) * state.conj(), dim=1)
+        # FIXME: cache this, it is very inefficient for non-parametric observables
+        block_mat = block_to_tensor(
```

Comment:
ok so we cant cache here?

Reply:
I haven't tried to fix this, it predates this PR. Not sure though how I could cache the tensor. Any suggestions ?

Reply:
the way it was before, which is a very recent MR of mine, did exactly that. in the constructor you check if its a parametric block, if not, you create the tensor and store it as a buffer. in the forward you check if the block is parametric and if not, use the buffer, otherwise build the tensor

Reply:
Alright, lemme have a look.

Reply:
Should be in 0c261c2.